lib/logstash/outputs/elasticsearch/ilm.rb

```diff
@@ -4,20 +4,22 @@ module Ilm
     ILM_POLICY_PATH = "default-ilm-policy.json"
 
     def setup_ilm
-      return unless ilm_enabled?
-      if default_index?(@index) || !default_rollover_alias?(@ilm_rollover_alias)
+      return unless ilm_in_use?
         logger.warn("Overwriting supplied index #{@index} with rollover alias #{@ilm_rollover_alias}") unless default_index?(@index)
         @index = @ilm_rollover_alias
         maybe_create_rollover_alias
         maybe_create_ilm_policy
-      end
     end
 
     def default_rollover_alias?(rollover_alias)
       rollover_alias == LogStash::Outputs::ElasticSearch::DEFAULT_ROLLOVER_ALIAS
     end
 
-    def ilm_enabled?
+    def ilm_alias_set?
+      default_index?(@index) || !default_rollover_alias?(@ilm_rollover_alias)
```

Comment:
why is this alias predicate negated? isn't the test about alias being set to `@ilm_rollover_alias`?

Reply:
It is about being set to something other than the default rollover alias